spec/01-unit/11-router_spec.lua

```diff
@@ -268,7 +268,47 @@ describe("Router", function()
       end)
     end)
 
-    describe("wildcard domains", function()
+    describe("[uri regex]", function()
+      it("matches with [uri regex]", function()
+        local use_case = {
+          {
+            name = "api-1",
+            uris = { [[/users/\d+/profile]] },
+          },
+        }
+
+        local router = assert(Router.new(use_case))
+
+        local api_t = router.select("GET", "/users/123/profile")
+        assert.truthy(api_t)
+        assert.same(use_case[1], api_t.api)
+      end)
+
+      it("matches the right API when several ones have a [uri regex]", function()
```

Comment:
This is the use case internally debated where the length of the regex URI does not have any meaning in terms of matching prioritization. In this case, because `/api/persons/\d{2}` was defined first, it will be evaluated first.